### PR TITLE
Port to zbus 4.0

### PR DESCRIPTION
**Pull request by bilelmoussaoui**

### CODE REVIEW

File: client/src/portal/secret.rs [hunk 1]
```diff
@@ -135,10 +134,7 @@ impl<'a> SecretProxy<'a> {
             async {
                 match self
                     .0
-                    .call_method(
-                        "RetrieveSecret",
-                        &(Fd::from(fd.as_fd().as_raw_fd()), &options),
-                    )
+                    .call_method("RetrieveSecret", &(Fd::from(fd.as_fd()), &options))
```
Comment (A6GibKm) [inline client/src/portal/secret.rs:137]: .call_method("RetrieveSecret", &(Fd::from(fd)), &options))

**Comment by A6GibKm** [comment]
Besides the ashpd lint in CI, looks ok to me. Still have not tested it.

File: client/src/portal/api/mod.rs [hunk 2]
```diff
@@ -219,7 +219,7 @@ impl Keyring {
 
         blob.push(MAJOR_VERSION);
         blob.push(MINOR_VERSION);
-        blob.append(&mut zvariant::to_bytes(*gvariant_encoding(), &self)?);
+        blob.append(&mut zvariant::to_bytes(*gvariant_encoding(), &self)?.to_vec());
```
[the remark this answers is not in the document]
Comment (A6GibKm) [reply client/src/portal/api/mod.rs:222]: alternatively, return a slice without allocating.

[the remark this answers is not in the document]
Comment (bilelmoussaoui) [reply client/src/portal/api/mod.rs:222]: Could be done separately, also this is private API so not that "bad"

**Pull request merged**